blocksync/block_stats.go

```diff
@@ -0,0 +1,98 @@
+package blocksync
+
+// blockStats is a circular buffer that holds at most n elements
+// and provides O(1) average calculation and O(1) max retrieval
+type blockStats struct {
+	buffer   []float64
+	capacity int
+	size     int     // current number of elements
+	head     int     // index where the next element will be written
```

Comment:
I'm thinking about `head`. Why not just do: 
```
	if len(rb.buffer) == rb.capacity {
		copy(rb.buffer, rb.buffer[1:])
		rb.buffer[len(rb.buffer)-1] = value
	} else {
       rb.buffer = append(rb.buffer, value)
    } 
```
instead of using `head` and all the `Add` logic. I might be missing something tho

Reply:
as per sync, this is not needed